Library/Homebrew/cmd/audit.rb

```diff
@@ -827,12 +827,30 @@ def audit_version
   def audit_checksum
     return unless checksum
 
-    case checksum.hash_type
-    when :md5
-      problem "MD5 checksums are deprecated, please use SHA1 or SHA256"
-      return
-    when :sha1   then len = 40
-    when :sha256 then len = 64
+    strict = ARGV.include? "--strict"
+    md5 = checksum.hash_type =~ /md5/
+    sha1 = checksum.hash_type =~ /sha1/
+    sha256 = checksum.hash_type =~ /sha256/
+
+    if strict
+      if md5
+        problem "MD5 checksums are deprecated, please use SHA256"
+        return
+      elsif sha1
+        problem "SHA1 checksums are deprecated, please use SHA256"
+        return
+      elsif sha256
+        len = 64
+      end
+    else
+      if md5
+        problem "MD5 checksums are deprecated, please use SHA256"
+        return
+      elsif sha1
+        len = 40
+      elsif sha256
+        len = 64
+      end
```

Comment:
@mikemcquaid suggested in a now-hidden comment (It's in one of the outdated diff sections) that we should scrap the ` case ` and move to ` if ` and ` elsif `. Happy to re-adopt the ` case ` if he has no objections.